apps/angular/crud/src/app/app.component.ts

```diff
@@ -1,51 +1,52 @@
 import { CommonModule } from '@angular/common';
-import { HttpClient } from '@angular/common/http';
-import { Component, OnInit } from '@angular/core';
-import { randText } from '@ngneat/falso';
-
+import { Component, OnInit, effect } from '@angular/core';
+import { TodoService } from './services/todo.service';
+import { todo } from './todo.model';
+import { MatProgressSpinnerModule } from '@angular/material/progress-spinner';
 @Component({
   standalone: true,
-  imports: [CommonModule],
+  imports: [CommonModule, MatProgressSpinnerModule],
   selector: 'app-root',
   template: `
-    <div *ngFor="let todo of todos">
+    <!-- 
+    // this is optional  
+    @if (isError) {
+      <p class="error">{{ isError }}</p>
+    } -->
+    @if (isLoading) {
+      <mat-spinner></mat-spinner>
+    }
+    <div *ngFor="let todo of todos$">
```

Comment:
I don't think this is working. ngFor is not imported. You should use the new control flow

Reply:
It was working, but now updated with @for